packages/react-native/ReactCommon/react/nativemodule/core/platform/ios/ReactCommon/RCTTurboModule.mm

```diff
@@ -400,12 +471,47 @@ id convertJSIValueToObjCObject(jsi::Runtime &runtime, const jsi::Value &value, s
       TurboModulePerfLogger::asyncMethodCallExecutionStart(moduleName, methodNameStr.c_str(), asyncCallCounter);
     }
 
-    @try {
+    try {
       [inv invokeWithTarget:strongModule];
-    } @catch (NSException *exception) {
-      throw convertNSExceptionToJSError(runtime, exception);
-    } @finally {
-      [retainedObjectsForInvocation removeAllObjects];
+    } catch (NSException *exception) {
```

Comment:
Can you make sure to test Objective-C exceptions? I'm not sure what the differences are between @catch and catch

Reply:
From what I tried -> Using `try-catch` catches both Objective-C `@throw`and `throw`, but `@try-@catch` only works for Objective-C and doesn't catch C++ (std::runtime_error).

Reply:
In other places in the codebase we've wrapped this as

```
try {
  @try {
     ...
   } @catch {
      // handle obj-c exceptions
   }
} catch {
  // handle c++
}